ecr/fetcher.go

```diff
@@ -77,13 +79,27 @@ func (f *ecrFetcher) Fetch(ctx context.Context, desc ocispec.Descriptor) (io.Rea
 }
 
 func (f *ecrFetcher) fetchManifest(ctx context.Context, desc ocispec.Descriptor) (io.ReadCloser, error) {
-	image, err := f.getImage(ctx)
+	var (
+		image *ecr.Image
+		err   error
+	)
+	// Single manifest images will not have a digest for the initial fetch, so
+	// its fetched by its ECRSpec reference. Otherwise, all fetches will be made
```

Comment:
Is this true?  If a caller specifies the image with a ref or name that contains both the tag and the digest you'll query for both.

Reply:
Yes, you're correct - a full specification provided and parsed [will include the digest, as given by ECRSpec](https://github.com/awslabs/amazon-ecr-containerd-resolver/pull/13/files#diff-c82d190de1354947418a3d1d2ee93a83R88). 

This needs rewording to reflect that, will update!